src/prototype.rs

```diff
@@ -766,11 +768,16 @@ fn parse_line(tree: &mut Tree<Item>, s: &str, mut ix: usize) -> (usize, Option<I
                 begin_text = ix;
             }
             b'[' => {
-                tree.append_text(begin_text, ix);
+                let (link_type, start) = if ix >= begin_text + 1 && bytes[ix - 1] == b'!' {
```

Comment:
This is currently causing a panic on spec_test_573:

```
\![foo]

[foo]: /url "title"
```

[Published as #564](https://spec.commonmark.org/0.28/#example-564)

Looking backwards by bytes in this "safe" ASCII-in-UTF-8 environment isn't quite as straightforward as looking forward by bytes.  In this case, if there are an odd number of `\` before the `!`, it's a link, but if there are an even number of `\` before, it's an image.

Reply:
Thanks for the heads-up! I hadn't considered that. I will try to issue a fix :-)